src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11114,7 +11071,43 @@ private ExpressionSyntax ParseExpressionContinued(ExpressionSyntax leftOperand,
                 }
 
                 // We'll "take" this operator, as precedence is tentatively OK.
-                var opToken = this.EatContextualToken(tk);
+
+                SyntaxToken opToken;
+
+                // Combine tokens into a single token if needed
+                if (tokensToCombine == 1)
+                {
+                    opToken = this.EatContextualToken(tk);
+                }
+                else if (tokensToCombine == 2)
+                {
+                    if (tk == SyntaxKind.DotToken)
+                    {
+                        opToken = this.EatDotDotToken();
+                    }
+                    else
+                    {
+                        opToken = SyntaxFactory.Token(
```

Comment:
nit: I think we should keep the same structure as original code and case below (for `tokensToCombine == 3`) using locals for the different parts. The inlining makes things harder to step through in the debugger.